### PR TITLE
Added parameter "mode" to ConfReport

### PR DESCRIPTION
**Pull request by areksolek**
Added parameter "mode" to ConfReport

### CODE REVIEW

**Comment by mariuszkrzaczkowski** [comment]
## Migration Recommended

It appears you are still using the CoPilot OAuth App - we recommend you [migrate to the GitHub App](https://github.com/apps/blackduck-copilot/installations/new/permissions?suggested_target_id=10201412&repository_ids[]=28060246)

**Comment by mariuszkrzaczkowski** [comment]
## Black Duck Security Report

Merging #11314 into developer will not change security risk.

### Added Components

Clean:       2

[Click here to see full report](https://copilot.blackducksoftware.com/github/repos/YetiForceCompany/YetiForceCRM/pull-requests/11314)

File: app/Utils/ConfReport.php [hunk 1]
```diff
@@ -1227,7 +1222,7 @@ public static function getErrors(string $type, bool $returnMore = false): array
 	{
 		$result = [];
 		foreach (static::get($type, true) as $param => $data) {
-			if (!$data['status']) {
+			if (!$data['status'] && (empty($data['mode']) || $data['mode'] === 0)) {
```
Comment (rskrzypczak) [inline app/Utils/ConfReport.php]: (empty($data['mode']) || $data['mode'] === 0)
it doesn't make sense

File: app/Utils/ConfReport.php [hunk 2]
```diff
@@ -281,7 +281,7 @@ class ConfReport
 		'user_privileges/' => ['type' => 'IsWritable', 'testCli' => true],
 		'user_privileges/tabdata.php' => ['type' => 'IsWritable', 'testCli' => true],
 		'user_privileges/menu_0.php' => ['type' => 'IsWritable', 'testCli' => true],
-		'user_privileges/user_privileges_1.php' => ['type' => 'IsWritable', 'testCli' => true, 'required'=>false],
+		'user_privileges/user_privileges_1.php' => ['type' => 'IsWritable', 'testCli' => true, 'mode'=> 4],
```
Comment (rskrzypczak) [inline app/Utils/ConfReport.php]:
```suggestion
		'user_privileges/user_privileges_1.php' => ['type' => 'IsWritable', 'testCli' => true, 'mode'=> self::MODE_{description}],
```
use constants in the file

File: layouts/basic/modules/Settings/ConfReport/Index.tpl [hunk 3]
```diff
@@ -16,7 +16,9 @@
 		{/if}
 	{/function}
 	{function HIGHLIGHT_ROW ITEM=[]}
-		{if !$ITEM['status'] && empty($ITEM['only_info'])}class="table-danger"{/if}
+		{if !$ITEM['status'] && (empty($ITEM['mode']) ||  $ITEM['mode'] eq 0)}class="table-danger"{/if}
+		{if !$ITEM['status'] && isset($ITEM['mode']) &&  $ITEM['mode'] eq 1}class="table-warning"{/if}
+		{if !$ITEM['status'] && isset($ITEM['mode']) &&  $ITEM['mode'] eq 2}class=""{/if}
```
Comment (rskrzypczak) [inline layouts/basic/modules/Settings/ConfReport/Index.tpl]: use elseif or function

File: app/Utils/ConfReport.php [hunk 4]
```diff
@@ -1134,9 +1134,6 @@ private static function validateIsWritable(string $name, array $row, string $sap
 	{
 		$row['status'] = \App\Fields\File::isWriteable($name);
 		$row[$sapi] = $row['status'] ? 'LBL_YES' : 'LBL_NO';
-		if(isset($row['required']) && $row['required'] === false && !file_exists(\ROOT_DIRECTORY . \DIRECTORY_SEPARATOR .$name)){
-			$row['skip'] = true;
```
Comment (rskrzypczak) [inline app/Utils/ConfReport.php]: ?

File: app/Utils/ConfReport.php [hunk 5]
```diff
@@ -281,7 +281,7 @@ class ConfReport
 		'user_privileges/' => ['type' => 'IsWritable', 'testCli' => true],
 		'user_privileges/tabdata.php' => ['type' => 'IsWritable', 'testCli' => true],
 		'user_privileges/menu_0.php' => ['type' => 'IsWritable', 'testCli' => true],
-		'user_privileges/user_privileges_1.php' => ['type' => 'IsWritable', 'testCli' => true, 'required'=>false],
+		'user_privileges/user_privileges_1.php' => ['type' => 'IsWritable', 'testCli' => true, 'mode'=> 'showErrors'],
```
Comment (rskrzypczak) [inline app/Utils/ConfReport.php]: ?

**Pull request merged**